spec/unit/mutant/meta/example/dsl_spec.rb

```diff
@@ -55,19 +49,29 @@ def self.expect_error(message, &block)
       end
     end
 
-    context 'on node that needs unparser preprocessing to be normalized' do
-      let(:node) { s(:send, s(:float, -1.0), :/, s(:float, 0.0)) }
-
-      let(:source) { '(-1.0) / 0.0' }
+    context 'using #declare lvar' do
+      let(:lvars)  { %i[a]        }
+      let(:node)   { s(:lvar, :a) }
+      let(:source) { 'a'          }
 
       expect_example do
-        source '(-1.0) / 0.0'
+        declare_lvar :a
+
+        source 'a'
       end
     end
 
     context 'using #mutation' do
-      let(:source)   { 'false'   }
-      let(:expected) { [s(:nil)] }
+      let(:source) { 'false' }
+
+      let(:expected) do
+        [
+          Mutant::Meta::Example::Expected.new(
+            node:            s(:nil),
+            original_source: 'nil'
+          )
+        ]
+      end
```

Comment:
This is kind of interesting. I'm not sure I fully understand the context of this spec, but does this mutate `false` -> `nil` somehow? I removed that mutation recently (I thought). Or is this just an artifact of test setup but wouldn't actually happen in practice?

Reply:
This spec does not demonstrate the mutations. It never calls the mutation engine, AKA `Mutant::Mutator::Node` is never called. Instead it specifies on how mutations are specified. For this I run through the scenarios:

* There was a mutation being generated that was not expected.
* A mutation that was expected was not being generated.
* A mutation that was generated unparses to invalid syntax
* The original code that was being specified did not round trip.

And for each of these cases above I set-up an `Mutant::Meta::Example` object in that specific state. But I intentionally do NOT run the engine here. But still use AST nodes that can round trip to unparser at low complexity.

Reply:
or TLDR: This code is the specification of the mutant mutation specifications and uses fake mutated nodes to not hard code the specification of properties of the specification against the mutation engine.

Reply:
> or TLDR: This code is the specification of the mutant mutation specifications and uses fake mutated nodes to not hard code the specification of properties of the specification against the mutation engine.

Ok, cool. I thought that might be the case but wasn't 100% sure at a glance. Might make sense to use a valid mutation for documentation consistency but probably doesn't matter much.

Reply:
>  Might make sense to use a valid mutation for documentation consistency but probably doesn't matter much.

it is a valid mutation example to demonstrate the specification of the specification on the properties I listed above. And I do not wish to call the real mutation engine here. I only wish to demonstrate I can find specific expected, missing and invalid mutations.

Reply:
Right, I just meant I usually like to have examples in my tests match "real" behavior as much as possible. It is clearly valid in this context to mutate to anything but I thought it might be slightly less confusing to match the "real" behavior. Doesn't actually matter though, feel free to ignore.

Reply:
If I where to start having to manually have generated mutations in sync here, I'd enforce something that adds no value to the test. And even would break as mutations are about to change drastically.

I'm experimenting a lot with "tiers" of mutations to default to only the most useful etc.